hack/verify-golangci-lint.sh

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+VERSION=v1.23.8
+URL=https://raw.githubusercontent.com/golangci/golangci-lint
+GOLANGCI_LINT=./bin/golangci-lint
+
+if [[ ! -f .golangci.yml ]]; then
+    echo 'ERROR: missing .golangci.yml in repo root' >&2
+    exit 1
+fi
+
+curl -sfL $URL/$VERSION/install.sh | sh -s $VERSION
```

Comment:
It seems we're unconditionally installing the tool every time we run the script?
We should do something like:
- check tool exists using `which`
- check tool is the correct version
- if tool doesn't exist, then download it
- run the tool

Reply:
I changed the script to:
- download only if `golangci-lint` is not in `$PATH`
- modify path if `golangci-lint` got downloaded